src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueSources.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Allows a user to create Syntax based input nodes for incremental generation
+    /// </summary>
+    public readonly struct SyntaxValueSources
+    {
+        private readonly PerGeneratorInputNodes.Builder _builder;
+
+        internal SyntaxValueSources(PerGeneratorInputNodes.Builder builder)
```

Comment:
>PerGeneratorInputNodes.Builder builder [](http://example.com/codeflow?start=36&length=38)

Could this be `ArrayBuilder<ISyntaxTransformNode> syntaxTransformNodes` instead?